libs/controllers/network/__init__.py

```diff
@@ -26,24 +28,34 @@ def __init__(self, type: int, message: bytes, source_address: int, destination_a
         self.data = message
 
     def serialize(self) -> bytes:
-        return b''.join([
+        frame = b''.join([
             self.type.to_bytes(1, 'big'),
             self.source_address.to_bytes(2, 'big'),
             self.destination_address.to_bytes(2, 'big'),
             self.data
         ])
 
+        return CRC().encode(frame)
```

Comment:
do we need to make a new instance every time we encode a message. 
if not would it be possible to move this to the INetworkController.send_message function